pkg/pubsub/config.go

```diff
@@ -2,38 +2,131 @@ package pubsub
 	Publisher struct {
-		MaxAttempts  int           `mapstructure:"max_attempts"`
+		// MaxAttempts represents the maximum number of times
+		// the client will try to send message again in case of failure
+		MaxAttempts int `mapstructure:"max_attempts"`
+		// WriteTimeout the maximum amount of time the client will wait for message to be written to Kafka topic
 		WriteTimeout time.Duration `mapstructure:"write_timeout"`
-		Topic        string        `mapstructure:"topic"`
-		Enabled      bool          `mapstructure:"enabled"`
+		// Topic the Kafka topic name to publish messages to
+		Topic string `mapstructure:"topic"`
+		// Enabled whether the publisher is enabled or not
+		Enabled bool `mapstructure:"enabled"`
 	}
 
 	Subscriber struct {
-		Topic   string `mapstructure:"topic"`
+		// Topic the Kafka topic name to retrieve messages from
+		Topic string `mapstructure:"topic"`
+		// GroupId the Kafka consumer group id
 		GroupId string `mapstructure:"group_id"`
-		Enabled bool   `mapstructure:"enabled"`
+		// Enabled whether the subscriber id enabled or not
+		Enabled bool `mapstructure:"enabled"`
+	}
+
+	TLS struct {
+		// Enabled whether the TLS connection is enabled or not
+		Enabled bool `mapstructure:"enabled"`
+
+		// Ca Root CA certificate
+		Ca string `mapstructure:"ca"`
+		// Cert is a PEM certificate string
+		Cert string `mapstructure:"cert_pem"`
+		// CertKey is a PEM key certificate string
+		CertKey string `mapstructure:"cert_pem_key"`
+		// Passphrase is used in case the private key needs to be decrypted
+		Passphrase string `mapstructure:"passphrase"`
+		// InsecureSkipVerify whether to skip TLS verification or not
+		InsecureSkipVerify bool `mapstructure:"insecure_skip_verify"`
+	}
+
+	SASL struct {
+		// Enabled whether the SASL connection is enabled or not
+		Enabled bool `mapstructure:"enabled"`
+
+		// Mechanism is a string representation of the SASL mechanism
+		// Currently, only "plain" and "aws_msk_iam" are supported
+		Mechanism string `mapstructure:"mechanism"`
+		// The username to authenticate Kafka requests
+		Username string `mapstructure:"username"`
+		// The password to authenticate Kafka requests
+		Password string `mapstructure:"password"`
+		// AWSMskIam AWS MSK IAM configuration
+		// To learn more visit AWS MSK documentation
+		// https://docs.aws.amazon.com/msk/latest/developerguide/iam-access-control.html
+		AWSMskIam SASLAwsMskIam `mapstructure:"aws_msk_iam"`
+	}
+
+	SASLAwsMskIam struct {
+		// AWS MSK IAM access key to authenticate AWS MSK requests
+		AccessKey string `mapstructure:"access_key"`
+		// AWS MSK IAM secret key to authenticate AWS MSK requests
+		SecretKey string `mapstructure:"secret_key"`
+
+		// SessionToken is used to authenticate AWS MSK requests via AWS STS service
+		// For more information see https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html
+		SessionToken string `mapstructure:"session_token"`
+		// UserAgent the client's user agent string
```

Comment:
nit: `The client's user agent string`

_(assuming we change the `README`, we should probably change this part too)_